src/ess/diffraction/filtering.py

```diff
@@ -67,3 +69,58 @@ def remove_bad_pulses(
     filtered = filtered.squeeze('good_pulse').copy(deep=False)
     del filtered.coords['good_pulse']
     return filtered
+
+
+def crop_tof(
+    data: RawData[RunType], tof_range: ValidTofRange
+) -> TofCroppedData[RunType]:
+    """Remove events outside the specified TOF range.
+
+    Parameters
+    ----------
+    data:
+        Data to be cropped.
+        Expected to have a coordinate called `'tof'`.
+    tof_range:
+        1d, len-2 variable containing the lower and upper bounds for
+        time-of-flight.
+
+    Returns
+    -------
+    :
+        Cropped data.
+    """
+    return TofCroppedData[RunType](
+        data.bin(tof=tof_range.to(unit=data.coords['tof'].unit))
+    )
+
+
+def filter_events(data: TofCroppedData[RunType]) -> FilteredData[RunType]:
+    """Remove bad events.
+
+    Attention
+    ---------
+    This function currently does nothing because it is unclear how to filter
+    events at ESS.
+    In the future, this function will filter out events that
+    cannot be used for analysis.
+
+    Parameters
+    ----------
+    data:
+        Input events to be filtered.
+
+    Returns
+    -------
+    :
+        `data` with bad events removed.
+    """
+    # TODO this needs to filter by proton charge once we know how
```

Comment:
Is this a new or old todo?

Reply:
It wasn't written down before. I added this provider in order to define a `FilteredData` type and make the tof-cropping explicit in the graph. Alternatively, I could drop this provider and make `crop_tof` return `FilteredData`.